antlr/src/main/antlr/FSH.g4

```diff
@@ -1,17 +1,17 @@
 grammar FSH;
 
 doc:                entity* EOF;
-entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem;
+entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet;
 
 alias:              KW_ALIAS SEQUENCE EQUAL SEQUENCE;
 
 profile:            KW_PROFILE SEQUENCE sdMetadata+ sdRule*;
 extension:          KW_EXTENSION SEQUENCE sdMetadata* sdRule*;
-sdMetadata:         parent | id | title | description;
+sdMetadata:         parent | id | title | description | mixins;
 sdRule:             cardRule | flagRule | valueSetRule | fixedValueRule | containsRule | onlyRule | obeysRule | caretValueRule;
 
 instance:           KW_INSTANCE SEQUENCE instanceMetadata* fixedValueRule*;
-instanceMetadata:   instanceOf | title | description;
+instanceMetadata:   instanceOf | title | description | mixins;
```

Comment:
If `instanceMetadata` allows mixins, and if mixins allow _all_ rule types, then _somewhere_ we need to add validation of the mixedin ruleset to ensure it only contains rule types supported by instances (currently FixedValueRules only).  Logically, it makes sense (to me) for that check to be on the import side -- but it would likely need to be a post-processing step since it's possible that an instance might mix in a RuleSet that hasn't been imported yet.  What do you think?

Reply:
I have this happening on the export side. I see why it makes sense on the import side, but let me make the case for the export side. The `Mixins` on an `Instance` don't get resolved to actual `RuleSet`s until the export side. On the import side the `Mixins` are just tracked as a list of strings that may or may not resolve to some actual `RuleSet`s. So to check on the import side, I would need to post-process as you said, and do the resolution to `RuleSet`s that is already happening on the export side, just to check the `RuleSet` has the correct types of rules, not to actually apply them. If I do this check on the export side, the `Mixin` name only has to be resolved to a `RuleSet` in this one place.

Additionally, I convinced myself it made sense to log errors on the export side, because these errors are about invalid rules, and the export side is where all other errors about invalid rules get logged. But maybe I convinced myself because it was more convenient to code 😉 , so if I haven't convinced you, let me know and I'll add the check on the import.

Reply:
OK.  I'm convinced.  Export side it is!